### DISCUSSION TITLE
Possibility to select cURL Client to enable concurrent requests

### DISCUSSION THREAD

**Discussion by bzrncev** [laravel/framework#36135] [Ideas] [answered]
Currently when I am using Http facade to fetch some API, the requests are synchronous, so if I made few requests I have to wait for the previous request to finish.

Symphony HTTP client have cURL support which allows [concurrent requests ](https://symfony.com/doc/current/http_client.html#concurrent-requests) by default and the following code makes 379 requests in less than half a second.

```php
$responses = [];
for ($i = 0; $i < 379; ++$i) {
    $uri = "https://http2.akamai.com/demo/tile-$i.png";
    $responses[] = $client->request('GET', $uri);
}

foreach ($responses as $response) {
    $content = $response->getContent();
    // ...
}
```

What do you think about adding this functionality to Laravel core?

**Comment by crynobone** [comment, marked as answer]
Should already be possible: https://laravel.com/docs/10.x/http-client#concurrent-requests
